rir/src/compiler/analysis/scope.cpp

```diff
@@ -136,6 +136,8 @@ AbstractResult ScopeAnalysis::doCompute(ScopeAnalysisState& state,
     } else if (Deopt::Cast(i)) {
         // who knows what the deopt target will return...
         state.returnValue.taint();
+        state.mayUseReflection = true;
```

Comment:
please document the dependencies between your PR's. this is from #499 right?

Reply:
yes